MonoGame.Framework/Vector3.cs

```diff
@@ -522,7 +530,7 @@ public bool Equals(Vector3 other)
         /// <returns>Hash code of this <see cref="Vector3"/>.</returns>
         public override int GetHashCode()
         {
-            return (int)(this.X + this.Y + this.Z);
+            return X.GetHashCode() + Y.GetHashCode() + Z.GetHashCode();
```

Comment:
> Hey are you serious ?

Yes.

You don't seem to understand what your doing is illegal.  You cannot look at the XNA code, change it slightly and submit it to MonoGame.  Just like you cannot copy a book someone wrote... change a few words around... and call it yours.